lib/kubeclient/common.rb

```diff
@@ -286,6 +308,45 @@ def get_entities(entity_type, klass, resource_name, options = {})
       Kubeclient::Common::EntityList.new(entity_type, resource_version, collection)
     end
 
+    def get_entities_async(klass, resource_name, options = {})
+      params = {}
+      SEARCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
+
+      ns_prefix = build_namespace_prefix(options[:namespace])
+
+      resource_version = nil
+      entity_streamer = Json::Streamer::JsonStreamer.new
+      resource_version_streamer = Json::Streamer::JsonStreamer.new
+
+      resource_version_streamer.get(key: 'resourceVersion') do |value|
+        resource_version = value unless resource_version
+      end
+
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      entity_streamer.get(nesting_level: 2, yield_values: false) do |object|
+        entity = new_entity(object, klass)
+        yield(entity, resource_version)
+      end
+
+      block = proc do |response|
+        if response.instance_of?(Net::HTTPNotFound)
+          raise ResourceNotFoundError.new(404,
+                                          'Not Found',
+                                          response)
+        end
```

Comment:
can we add a test for the negative flow?

Reply:
added some tests to handle exceptions / bad status codes